nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -170,8 +178,49 @@ public void shouldFailWhenRetrievingDocumentBlobWithoutColdStorageContent() {
         }
     }
 
-    protected DocumentModel createDocument(boolean addBlobContent) {
-        DocumentModel documentModel = session.createDocumentModel("/", "anyFile", "File");
+    @Test
+    public void shouldCheckAvailabilityOfColdStorageContent() {
+        List<String> documents = Arrays.asList( //
+                moveAndRetrieveColdStorageContent(DEFAULT_DOC_NAME).getId(),
+                moveAndRetrieveColdStorageContent("anyFile2").getId(), //
+                moveAndRetrieveColdStorageContent("anyFile3").getId());
+        transactionalFeature.nextTransaction();
+        try (CapturingEventListener listener = new CapturingEventListener(
+                ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME)) {
+            ColdStorageHelper.ColdStorageContentStatus coldStorageContentStatus = ColdStorageHelper.checkAvailabilityOfColdStorageContent(
+                    session);
+            assertEquals(session.getRepositoryName(), coldStorageContentStatus.getRepositoryName());
+
+            // all cold storage contents being retrieved are now available
+            assertEquals(documents.size(), coldStorageContentStatus.getTotalBeingRetrieved());
+            assertEquals(documents.size(), coldStorageContentStatus.getTotalContentAvailable());
+
+            assertTrue(listener.hasBeenFired(ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME));
+            assertEquals(3, listener.streamCapturedEvents().count());
+
+            List<String> docEvents = listener.streamCapturedEvents() //
+                                             .map(event -> {
+                                                 DocumentEventContext docCtx = (DocumentEventContext) event.getContext();
+                                                 return docCtx.getSourceDocument().getId();
+                                             }) //
+                                             .sorted() //
+                                             .collect(Collectors.toList());
+
+            documents.sort(Comparator.naturalOrder());
+            assertEquals(documents, docEvents);
+        }
+    }
+
+    protected DocumentModel moveAndRetrieveColdStorageContent(String documentName) {
+        DocumentModel documentModel = createDocument(documentName, true);
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+        ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                NUMBER_OF_DAYS_OF_AVAILABILITY);
+        return documentModel;
```

Comment:
Id rather see:
```
        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
        return ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
                NUMBER_OF_DAYS_OF_AVAILABILITY);
```
Since the methods return the DocumentModel we may as well use it.

Reply:
yes, i did it to avoid any false positive test assertion. as the helper set the value if i check only the returned doc as it set only this boolean, necessarily it will be `true` it's why i reload the doc and wait until the commit of `TX`

Reply:
As you wish, you can keep the next transaction call in the caller to ensure false positive detection, but it still feels weird in this method to return the initial document passed as a parameter instead of the updated one.

Reply:
i agree  sorry i will return directly the document in this case